internal/command/utils/lifecycle/before.go

```diff
@@ -83,18 +122,10 @@ func SetupClientWithPassphrase(c *cli.Context) error {
 	return nil
 }
 
-func UnlockAccount(c *cli.Context) error {
-	client := c.Context.Value(core.ClientKey).(*client.Client)
-
-	passphrase, err := prompt.AccountPassphrase().RunFlag(c, "passphrase")
-	if err != nil {
-		return err
-	}
-
-	err = client.Signer().Unlock(client.Account(), passphrase)
-	if err != nil {
+func SetupClientWithPassphrase(c *cli.Context) error {
+	if err := SetupClient(c); err != nil {
 		return err
 	}
 
-	return nil
+	return UnlockAccount(c)
```

Comment:
I would consider just moving the UnlockAccount logic here instead if its never called anywhere else.

Reply:
I originally had that logic here before, but since we are calling UnlockAccount separately on the commands that require it, splitting it out allows for setting up the client in read-only mode for commands like `org fetch` while unlocking the account for `org create` etc. On the `publish` command, it uses lifecycle.SetupClientWithPassphrase to do everything in one step vs SetupClient then UnlockAccount